qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java

```diff
@@ -0,0 +1,115 @@
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.JobBatchIntent;
+import io.camunda.zeebe.qa.util.testcontainers.ZeebeTestContainerDefaults;
+import io.camunda.zeebe.test.util.record.RecordStream;
+import io.camunda.zeebe.test.util.socket.SocketUtil;
+import io.camunda.zeebe.test.util.testcontainers.ContainerLogsDumper;
+import io.zeebe.containers.cluster.ZeebeCluster;
+import io.zeebe.containers.engine.ContainerEngine;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.StreamSupport;
+import org.agrona.CloseHelper;
+import org.assertj.core.groups.Tuple;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.Network;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+final class JobPushIT {
+  private final Network network = Network.newNetwork();
+  private final ZeebeCluster cluster =
+      ZeebeCluster.builder()
+          .withBrokersCount(1)
+          .withGatewaysCount(1)
+          .withPartitionsCount(1)
+          .withEmbeddedGateway(false)
+          .withImage(ZeebeTestContainerDefaults.defaultTestImage())
+          .build();
+
+  @SuppressWarnings("unused")
+  @RegisterExtension
+  final ContainerLogsDumper logsWatcher = new ContainerLogsDumper(cluster::getNodes);
+
+  @Container
+  private final ContainerEngine engine =
+      ContainerEngine.builder()
+          .withDebugReceiverPort(SocketUtil.getNextAddress().getPort())
+          .withCluster(cluster)
+          .withIdlePeriod(Duration.ofSeconds(5))
+          .build();
+
+  @AfterEach
+  void afterEach() {
+    CloseHelper.quietCloseAll(network);
+  }
+
+  @Test
+  void shouldStreamActivatedJobs() throws InterruptedException, TimeoutException {
+    // given
+    final var process =
+        Bpmn.createExecutableProcess("process")
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType("foo"))
+            .endEvent()
+            .done();
+
+    try (final var client = engine.createClient()) {
+      final var deploymentEvent =
+          client.newDeployResourceCommand().addProcessModel(process, "process.bpmn").send().join();
+
+      // when
+      final List<ActivatedJob> jobs = new ArrayList<>();
+      client
+          .newStreamJobsCommand()
+          .jobType("foo")
+          .consumer(jobs::add)
+          .timeout(Duration.ofMinutes(2))
+          .send();
+
+      client
+          .newCreateInstanceCommand()
+          .processDefinitionKey(deploymentEvent.getProcesses().get(0).getProcessDefinitionKey())
+          .requestTimeout(Duration.ofMinutes(1))
+          .send()
+          .join();
```

Comment:
:x: This section is flaky for two reasons:

1. We can only push jobs to a stream if the stream existed before the job handled (i.e. created, timed out, failed, etc. - anything that would make it activate-able again).
2. Opening a stream is asynchronous, so there's a chance the process instance is created, and the job is created, before the stream is opened.

There are some options to work around this here, but your best bet is to wait for my PR with actuators. Then you can actually query the broker to ensure the stream exists before creating the process instance.

> **Note**
> In production, since users will be using a job worker, this is not a big deal since we're both streaming and polling.

Reply:
As this may block you, we can do the other changes for this PR, and then I'm happy to take adapt your PR once mine is merged so you can already do other stuff.

Reply:
Alright, these makes sense. Then, let's wait for the blocking PR to be merged. Afterwards, please feel free to add changes.